packages/shell-api/src/collection.ts

```diff
@@ -2383,34 +2385,69 @@ export default class Collection extends ShellApiWithMongoClass {
     );
   }
 
+  async createSearchIndex(
+    name?: string,
+    definition?: SearchIndexDefinition
+  ): Promise<string>;
+  async createSearchIndex(
+    name?: string,
+    type?: 'search' | 'vectorSearch',
+    definition?: SearchIndexDefinition
+  ): Promise<string>;
+  async createSearchIndex(
+    definition?: SearchIndexDefinition,
+    type?: 'search' | 'vectorSearch'
+  ): Promise<string>;
+  async createSearchIndex(
+    description?: SearchIndexDescription
+  ): Promise<string>;
   @serverVersions(['6.0.0', ServerVersions.latest])
   @returnsPromise
   @apiVersions([])
-  // TODO(MONGOSH-1471): use SearchIndexDescription once available
   async createSearchIndex(
-    indexName?: string | Document,
-    type?: 'search' | 'vectorSearch' | Document,
-    definition?: Document
+    nameOrOptions?: string | SearchIndexDescription | SearchIndexDefinition,
+    typeOrOptions?: 'search' | 'vectorSearch' | SearchIndexDefinition,
+    definition?: SearchIndexDefinition
   ): Promise<string> {
-    if (typeof type === 'object' && type !== null) {
-      definition = type;
-      type = undefined;
+    let indexName: string | undefined;
+    let indexType: string | undefined;
+
+    if (typeof typeOrOptions === 'object' && typeOrOptions !== null) {
+      definition = typeOrOptions;
+    } else {
+      indexType = typeOrOptions;
     }
-    if (typeof indexName === 'object' && indexName !== null) {
-      definition = indexName;
-      indexName = undefined;
+
+    if (
+      typeof nameOrOptions === 'object' &&
+      nameOrOptions !== null &&
+      nameOrOptions.definition
+    ) {
+      indexName = nameOrOptions.name;
+      indexType = nameOrOptions.type;
+      definition = nameOrOptions.definition;
```

Comment:
If the user provides a SearchIndexDescription, we should just pass that value through instead of decomposing it and then re-composing it into a SearchIndexDescription